pkg/controller/user-cluster-controller-manager/resources/resources/mla/prometheus/deployment.go

```diff
@@ -78,7 +78,7 @@ func DeploymentCreator(overrides *corev1.ResourceRequirements, registryWithOverw
 			deployment.Spec.Selector = &metav1.LabelSelector{
 				MatchLabels: controllerLabels,
 			}
-			deployment.Spec.Replicas = pointer.Int32Ptr(1)
+			deployment.Spec.Replicas = pointer.Int32Ptr(2)
```

Comment:
So do we want to deploy 2 Prometheus replicas unconditionally for every user cluster? I think this should be rather somehow configurable.

Reply:
I have to admit I am not sure. So from a technical perspective the resource-overhead is comparatively small. It basically consists of:
1. running one more instance of prometheus + egress to cortex distributor for this instance
2. number more cortex distributors scales slightly higher with the number of user clusters. Number of cortex ingestors will stay the same.

From a product/customer perspective I am not deep enough into the topic and how customers use it yet. 
Hope the technical perspective helped a bit, maybe someone else can give some insights into the product/customer view

Reply:
2 replicas probably would be the best default option, to achieve HA. But I can also imagine small resource-constrained clusters where 1 would be preferred (even if it can mean some temporary data loss) or big clusters where even 3 would make sense. But I don't have any strong opinion either.

Another thing to consider is: Imagine we have an old KKP installation with user cluster MLA running. After KKP upgrade, all user clusters will suddenly deploy 2 Prometheus replicas instead of one. Will this cause any issues if the cortex backend in Seed is not upgraded?

Reply:
@SimonTheLeg @rastislavs All good points, maybe let's keep it as 2 default, and we make this configurable via KKP API in the next PR? I can create a user story for that